nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -0,0 +1,339 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.mastodon;
+
+  env = {
+    RAILS_ENV = "production";
+    NODE_ENV = "production";
```

Comment:
Mild disagree. I’m not sure running Mastodon through a NixOS-generated systemd service that has to be rebuilt every time you make a change is a reasonable development flow.

Reply:
Shouldn't a user be able to manually do that via
```nix
systemd.services.mastodon-init-db.environment.RAILS_ENV = lib.MkForce "development";
systemd.services.mastodon-streaming.environment.RAILS_ENV = lib.MkForce "development";
systemd.services.mastodon-web.environment.RAILS_ENV = lib.MkForce "development";
systemd.services.mastodon-sidekiq.environment.RAILS_ENV = lib.MkForce "development";
```
(or a subset of that) anyways?

Reply:
iI'm not sure if this would have a higher priority than EnvironmentFile.
But now that you mention it, I would like to expose the environment variables as a NixOS option.

Reply:
You can now use `services.mastodon.extraOptions.RAILS_ENV = "development";`